src/main/java/core/AbstractGameState.java

```diff
@@ -163,7 +172,17 @@ public final AbstractGameState copy(int playerId) {
      * @param playerId - player observing the state.
      * @return - double, score of current state.
      */
-    protected abstract double _getScore(int playerId);
+    protected abstract double _getHeuristicScore(int playerId);
+
+    /**
+     * This provides the current score in game turns. This will only be relevant for games that have the concept
+     * of victory points, etc.
+     * If a game does not support this directly, then just return 0.0
+     *
+     * @param playerId
+     * @return - double, score of current state
+     */
+    public abstract double getGameScore(int playerId);
```

Comment:
I'm wondering if there are more games that have a scoring systems than those that don't. If so, having this method abstract and forcing all games to implement it is fine. But forcing all games to implement it just to return 0 in a majority is not ideal, and could easily just be done in the core code (and any other games that do have scoring can overwrite it). Maybe something to revise at some point.

Reply:
My view on this was that a score is a very common feature in games (not, admittedly always so, as Exploding Kittens shows). I felt that it was always worth asking the implementor of a game to ask themselves this question, especially as if the answer is 'No', then we have clear javadoc to tell them what to put in as a default...and it shouldn't take them more than a second or two to write "return 0.0".
Having a default implementation to do this I felt could be dangerous if future implementors don;t realise they *should* ask themslves this question.